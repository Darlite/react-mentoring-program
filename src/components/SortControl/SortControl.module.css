.sortControl {
    display: flex;
    gap: 20px;
    text-transform: uppercase;
}

.sortOrderButton {
    background: none;
    border: none;
    cursor: pointer;
}

.arrowUp {
    display: inline-block;
    width: 0;
    height: 0;
    border-top: 0;
    border-bottom: 10px solid #F65261;
    border-left: 10px solid transparent;
    border-right: 10px solid transparent;
    border-radius: 5px;
}

.arrowDown {
    display: inline-block;
    width: 0;
    height: 0;
    border-bottom: 0;
    border-top: 10px solid #F65261;
    border-left: 10px solid transparent;
    border-right: 10px solid transparent;
    border-radius: 5px;
}
