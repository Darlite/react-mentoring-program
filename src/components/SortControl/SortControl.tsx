import React from "react";
import styles from "../../styles/SortControl.module.css";

interface SortControlProps {
    currentSelection: string;
    onSelect: (option: string) => void;
}

const SortControl: React.FC<SortControlProps> = ({currentSelection, onSelect}) => {
    return (
        <div className={styles.sortControl}>
            <label>Sort By</label>
            <select onChange={(event) => onSelect(event.target.value)}
                    value={currentSelection}>
                <option value="Release Date">Release Date</option>
                <option value="Title">Title</option>
            </select>
        </div>
    )
}

export default SortControl;
