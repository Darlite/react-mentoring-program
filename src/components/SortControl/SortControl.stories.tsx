import {Meta, StoryObj} from "@storybook/react";
import {action} from "@storybook/addon-actions";
import SortControl from "./SortControl";
import {fn} from "@storybook/test";

const meta = {
    component: SortControl,
    title: 'SortControl',
    tags: ['autodocs'],
    argTypes: {
        currentSelection: {
            control: {type: "select"},
            options: ["Release Date", "Title"],
        }
    },
    args: {
        currentSelection: "Release Date",
    }
} satisfies Meta<typeof SortControl>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {
    args: {
        currentSelection: "Release Date",
        onSelect: action("onSelect"),
        sortOrder: "asc",
        onSortOrderChange: fn(),
    },
};
