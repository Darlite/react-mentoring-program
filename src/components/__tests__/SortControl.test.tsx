import {render, screen} from "@testing-library/react";
import SortControl from "../SortControl/SortControl";
import userEvent from "@testing-library/user-event";

describe('SortControl', () => {
    it("renders correctly", () => {
        const { container } = render(<SortControl currentSelection={"Release Date"} onSelect={() => {}} />);
        expect(container).toMatchSnapshot();
    });

    it("renders with initial selection", () => {
        const onSelect = jest.fn();
        render(<SortControl currentSelection={"Release Date"} onSelect={onSelect} />);
        const sortControlSelect = screen.getByRole("combobox");
        expect(sortControlSelect).toHaveValue("release_date");
    });

    it("handles the other selection", () => {
        const onSelect = jest.fn();
        render(<SortControl currentSelection={"Release Date"} onSelect={onSelect} />);
        const sortControlSelect = screen.getByRole("combobox");
        userEvent.selectOptions(sortControlSelect, "Title");
        expect(onSelect).toHaveBeenCalledWith("title");
    });
})