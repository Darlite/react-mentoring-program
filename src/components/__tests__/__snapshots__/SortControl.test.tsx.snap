// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortControl renders correctly 1`] = `
<div>
  <div
    class="sortControl"
  >
    <label>
      Sort By
    </label>
    <select>
      <option
        value="Release Date"
      >
        Release Date
      </option>
      <option
        value="Title"
      >
        Title
      </option>
    </select>
  </div>
</div>
`;
